Algorithms/Search/RecursiveBinarySearcher.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Algorithms.Searches
+{
+    /// <summary>
+    /// TODO.
+    /// </summary>
+    /// <typeparam name="T">TODO. 2.</typeparam>
+    public class RecursiveBinarySearcher<T> where T : IComparable<T>
+    {
+        public static readonly int NotFoundIndex = -1;
```

Comment:
Codacy found an issue: [A static field in a generic type is not shared among instances of different close constructed types.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6286772)

Reply:
Maybe making it a constant helps?